[project]
name = "moehlenhoff-alpha2"
version = "1.4.0"
description = "Python client for the Moehlenhoff Alpha2 underfloor heating system"
readme = "README.md"
classifiers = [
  "License :: OSI Approved :: GNU General Public License v3 (GPLv3)",
]
requires-python = ">=3.10"
dependencies = [
  "aiohttp",
  "xmltodict",
]

[[project.authors]]
name = "Jan Schneider"
email = "oss@janschneider.net"

[project.urls]
Homepage = "https://github.com/j-a-n/python-moehlenhoff-alpha2"
Repository = "https://github.com/j-a-n/python-moehlenhoff-alpha2"

[dependency-groups]
dev = [
  "mypy>=1.14.1",
  "pytest>=7.1",
  "pytest-asyncio>=0.18",
  "pytest-coverage>=0.0",
  "ruff>=0.11.0",
  "types-xmltodict>=0.14.0.20241009",
]

[tool.uv]
package = true

[tool.ruff]
line-length = 140
target-version = "py310"

[tool.mypy]
disallow_untyped_calls = true
disallow_untyped_defs = true
disallow_incomplete_defs = true
follow_imports = "silent"

[tool.pytest.ini_options]
asyncio_mode = "auto"
asyncio_default_fixture_loop_scope = "function"

# FIXME: temporary workaround until https://github.com/pypa/setuptools/issues/4759 is fixed
[tool.setuptools]
license-files = []
